zerver/views/registration.py

```diff
@@ -1386,6 +1391,149 @@ def create_realm(request: HttpRequest, confirmation_key: str | None = None) -> H
     )
 
 
+def generate_demo_realm_subdomain() -> str:
+    available_subdomain = False
+    demo_subdomain = ""
+    while not available_subdomain:
+        letters = "".join(random.SystemRandom().choice(string.ascii_lowercase) for _ in range(4))
+        digits = "".join(random.SystemRandom().choice(string.digits) for _ in range(4))
+        demo_subdomain = f"demo-{letters}{digits}"
+        try:
+            check_subdomain_available(demo_subdomain)
+            available_subdomain = True
```

Comment:
This could be `break`, and then the loop a `while True`, if one wants to remove the `available_subdomain` variable and make it a bit shorter. Maybe this is nicely readable, though.